File: programmers/js/level00/120864.js
// 숨어있는 숫자의 덧셈 (2)
function solution(my_string) {
  const nums = my_string.match(/\d+/g);
    
  return nums ? nums.reduce((acc, cur) => acc + +cur, 0) : 0;
}

// 다른 풀이
// function solution(my_string) {
//   return my_string.split(/\D+/).reduce((acc, cur) => acc + Number(cur), 0);
// }

File: programmers/java/level00/120864.java
// 숨어있는 숫자의 덧셈 (2)

import java.util.Arrays;

class Solution {
    public int solution(String my_string) {
        return Arrays.stream(my_string.split("[A-Za-z]"))
            .filter(str -> str.length() > 0)
            .mapToInt(Integer::valueOf)
            .sum();
    }
}